Examples/Holocron-iOSExample/ViewController.swift

```diff
@@ -7,127 +7,122 @@
 
 import UIKit
 import Holocron
-import Result
+
+extension Key {
+    static let fileTest: Key = "fileTestKey"
+    static let keychainTest: Key = "testKeychainKey"
+    static let userDefaultsTest: Key = "testUserDefaultsKey"
+}
 
 class ViewController: UIViewController {
-    
-    lazy var userDefaultsPersistence: UserDefaultsPersistence = UserDefaultsPersistence()
-    let userDefaultsStore = UserDefaultsStore(key: "testUserDefaults")
-    
-    lazy var keychainPersistence: KeychainPersistence = KeychainPersistence(keychainServiceName: "com.holocron.test")
-    let keyChainStore = KeychainStore(key: "testKeychainPersistence")
-    
-    let cacheDirectory = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first!
-    lazy var filePersistance: FilePersistence = FilePersistence(directory: cacheDirectory)
-    let fileStore = FileStore(fileName: "testFilePersistance")
-    
+    let defaultsProvider: StorageProvider = UserDefaults.standard
+    let keychainProvider: StorageProvider = .Keychain(service: "com.holocron.example")
```

Comment:
That actually is a type. The `.Keychain` you're seeing is a typealias nested in `StorageProvider` that aliases `KeychainAccess.Keychain`.

Reply:
I get how it works, I guess I was just saying it feels a little bit non-standard since you usually only see that style of type inference for enums and static properties (both of which are lowercase following the `.`). We can leave it how it is unless it catches anyone else's eye.